git-imerge

```diff
@@ -3640,8 +3653,27 @@ def cmd_reparent(parser, options):
     except ValueError as e:
         sys.exit(e.message)
 
-    sys.stdout.write('%s\n' % (git.reparent(commit_sha1, parent_sha1s),))
+    iterative_reparent(head_sha1, commit_to_reparent_sha1, parent_sha1s, git)
+
+def iterative_reparent(head, commit_to_reparent, parents, git):
+    commit_list = [head];
+    current_commit = head;
+    while current_commit != commit_to_reparent:
+        current_parents = git.get_commit_parents(current_commit)
+        first_parent_commit = current_parents[0]
+        commit_list.append(first_parent_commit)
+        current_commit = first_parent_commit
+
+    commit_list.reverse()
+    for i in range(0, len(commit_list)):
+        if i == 0:
+            current_parents = parents
+        else:
+            current_parents = git.get_commit_parents(commit_list[i])
+            current_parents = [new_commit] + current_parents[1:len(current_parents)]
```

Comment:
This is going to do something screwy if there is more than one ancestry path between `commit_to_reparent` and `head`. For example, suppose that there was a merge in between:

```
o---A---C---D---E---F---H
       /     \     /
  o---B       X---Y
```

and you run `git imerge reparent --commit-to-reparent C A`. This algorithm will only rewrite commits `C`, `D`, `E`, `F`, and `H`, but won't rewrite `X` or `Y`. The result will not be the hoped-for

```
o---A---C'--D'--E'--F'--H'
             \     /
              X'--Y'
```

but I think will look like this instead:

```
      C'--D'--E'------F'--H'
     /               /
o---A---C---D       /
       /     \     /
  o---B       X---Y
```

I understand that you are not interested in this case, but it's not that much harder to handle it correctly.